### PR TITLE
 [Frontend] Set up CI: lint, format, test, build, security

### LINKED ISSUE #26: Create CI for the frontend
## Feature Description
Create a GH CI Workflow that runs on every `push` for the frontend that runs: linting, formatting, testing, security

## Problem You're Trying to Solve
We want to enforce that only code that "passes" tests/lints/etc are merged. Creating a CI workflow will ensure that "bad" code does not get merged and doesn't disrupt other developers.

### PR DESCRIPTION
**Pull request by MNThomson**
# Description

Create the base CI jobs for the Frontend service: linting, formatting, testing, building, and security

This PR is stacked on top of #87, so first review that PR (as this diff contains that PR diff also).

Closes #26

## How to Test

View the CI output below or run the actions locally with [act](https://github.com/nektos/act)

## Checklist
- [x] The code includes tests if relevant
- [x] I have *actually* self-reviewed my changes and done QA

### CODE REVIEW

**Review by TobySMurray — APPROVED** [review]
LGTM

File: apps/frontend/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import "./App.css";
+
+function App() {
+  return <h1>Martletplace</h1>;
```
Comment (sejal-agarwal) [inline apps/frontend/src/App.tsx:4]: MartletPlace pls, sorry to be nitpicky

**Review by sejal-agarwal — APPROVED** [review]
Appreciate you Max!!

**Review by CaelumD25 — APPROVED** [review]

**Review by ChloeZacharias — APPROVED** [review]

File: apps/frontend/src/App.tsx (hunk 1 above)
Comment (MNThomson) [reply apps/frontend/src/App.tsx:4]: My assumption is that wouldn't be capitalized...no?
image

**Pull request merged**